src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx

```diff
@@ -38,51 +40,94 @@ const FilterGroup: React.FC<FilterGroupProps> = ({
 
   const [newFilterName, setNewFilterName] = useState(name);
   const [isEditing, setIsEditing] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+  const editFieldRef = useRef<HTMLInputElement>(null);
 
   const resetEditState = () => {
     setIsEditing(false);
     setNewFilterName(name);
   };
 
+  const handleSubmit = () => {
+    if (isValid) {
+      editFilter("name", newFilterName, filter);
+      resetEditState();
+    }
+  };
+
+  useEffect(() => {
+    if (isEditing) {
+      editFieldRef.current?.focus();
+    }
+  }, [isEditing, editFieldRef]);
+
   return (
     <FilterContainer data-cy={dataCy}>
       <FilterHeader>
         <Badge>FILTER</Badge>
         <IconButtonContainer>
-          <IconButton
-            aria-label="Edit filter"
-            onClick={() => setIsEditing(true)}
+          <Tooltip
```

Comment:
I'm not sure if I like these tooltips ... Namely, I don't like how they show up on every single filter because they start to get distracting. I think the icons are pretty self-explanatory, and I'm not sure that we actually need the tooltips.

Reply:
This could be a good use case for a simple `title` prop added to `IconButton` since these have a slight delay before appearing

Reply:
Great point, I do like the title prop as a middle ground.